services/submission/src/main/java/app/coronawarn/server/services/submission/config/SubmissionServiceConfig.java

```diff
@@ -0,0 +1,65 @@
+package app.coronawarn.server.services.submission.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+@Component
+@EnableConfigurationProperties
+@ConfigurationProperties(prefix = "services.submission")
+public class SubmissionServiceConfig {
+
+  // Exponential moving average of the last N real request durations (in ms), where
+  // N = fakeDelayMovingAverageSamples.
+  private Double initialFakeDelayMilliseconds;
+  private Double fakeDelayMovingAverageSamples;
+  private Integer retentionDays;
+  private Payload payload;
+
+  public Double getInitialFakeDelayMilliseconds() {
+    return initialFakeDelayMilliseconds;
+  }
+
+  public void setInitialFakeDelayMilliseconds(Double initialFakeDelayMilliseconds) {
+    this.initialFakeDelayMilliseconds = initialFakeDelayMilliseconds;
+  }
+
+  public Double getFakeDelayMovingAverageSamples() {
+    return fakeDelayMovingAverageSamples;
+  }
+
+  public void setFakeDelayMovingAverageSamples(Double fakeDelayMovingAverageSamples) {
+    this.fakeDelayMovingAverageSamples = fakeDelayMovingAverageSamples;
+  }
+
+  public Integer getRetentionDays() {
+    return retentionDays;
+  }
+
+  public void setRetentionDays(Integer retentionDays) {
+    this.retentionDays = retentionDays;
+  }
+
+  public Integer getMaxNumberOfKeys() {
+    return payload.getMaxNumberOfKeys();
```

Comment:
One more design decision that needs critical review/thought.

Idea was to abstract away complexities (additional classes/getters) in configuration structure and only expose single contract outside.

Applies to `DistributionServiceConfig` as well.

Reply:
When abstracting the payload away, you will lose the information that `getMaxNumberOfKeys` refers to the payload. Then you would need to rename this to `getMaxNumberOfPayloadKeys`. Not sure if this is worth it. But if there is some redundant more complex operations done on the properties across the code, then this could be moved to this class.

Maybe [@ConstructorBinding](https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-external-config-constructor-binding) should be used to prevent in incomplete object that could result in NPEs (e.g. in `return payload.getMaxNumberOfKeys();).

Also, `@ConfigurationProperties` can be annotated with `@Validated` which will actually crash the app at start-up time with a proper readable error message in case non-optional properties are missing:

https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-external-config-validation